### PR TITLE
macOS UI Improvements

### PR DESCRIPTION
**Pull request by AvdLee**
> [!NOTE]  
> My previous PR should first be merged. This PR contains changes from that PR (#273). @kean ideally, I'd become a contributor so I can open PRs with different target branches. Let me know what you think!

This PR adds several UI improvements:

- [x] Add helpers to toolbar buttons
CleanShot 2024-07-01 at 09 09 59@2x
- [x] Updated the documentation regarding the mode (see inline comment)
- [x] Hide list scroll background to allow window background to shine through. This enables more customization options.
- [x] Update the console min width to better display network requests on macOS
- [x] Remove window style for macOS demo since the view looked broken:
CleanShot 2024-07-01 at 09 47 56@2x

Fixes https://github.com/AvdLee/RocketSimApp/issues/530

### CODE REVIEW

File: Sources/PulseUI/Features/Console/ConsoleView-macos.swift [hunk 1]
```diff
@@ -51,7 +51,7 @@ private struct ConsoleMainView: View {
 
     private var contentView: some View {
         ConsoleListView()
-            .frame(minWidth: 200, idealWidth: 400, minHeight: 120, idealHeight: 480)
+            .frame(minWidth: 400, idealWidth: 500, minHeight: 120, idealHeight: 480)
```
Comment (AvdLee) [inline Sources/PulseUI/Features/Console/ConsoleView-macos.swift:54]: With a width of 200, requests became unreadable and unusable. A minimum width of 400 works much better

File: Sources/PulseUI/Features/Console/ConsoleView.swift [hunk 2]
```diff
@@ -12,9 +12,9 @@ extension ConsoleView {
     ///
     /// - parameters:
     ///   - store: The store to display. By default, `LoggerStore/shared`.
-    ///   - mode: The console mode. By default, ``ConsoleMode/all``. If you change
-    ///   the mode to ``ConsoleMode/network``, the console will only display the
-    ///   network messages.
+    ///   - mode: The initial console mode. By default, ``ConsoleMode/all``. If you change
+    ///   the mode to ``ConsoleMode/network``, the console will display the
+    ///   network messages up on appearance.
```
Comment (AvdLee) [inline Sources/PulseUI/Features/Console/ConsoleView.swift:17]: Correct me if I misunderstood, but the original documentation made me think I could completely hide the `All` and `Logs` tab buttons. Looking at the code and final outcome it turned out to only configure the initial mode. Hence why I updated the documentation.

File: Sources/PulseUI/Features/Console/List/ConsoleListView.swift [hunk 3]
```diff
@@ -174,7 +174,7 @@ private struct _ConsoleListView: View {
                     } else {
                         ConsoleListContentView(proxy: proxy)
                     }
-                }
+                }.scrollContentBackground(.hidden)
```
Comment (AvdLee) [inline Sources/PulseUI/Features/Console/List/ConsoleListView.swift:177]: This makes the view on macOS more beautiful in my opinion and also allows to configure an overall console background color using:

```swift
ConsoleView()
    .backgroundColor(Color.black)
```

For example.

**Comment by kean** [comment]
>  I'd become a contributor so I can open PRs with different target branches. Let me know what you think!

Absolutely, I sent an invite!

File: Sources/PulseUI/Features/Console/ConsoleView.swift (hunk 2 above)
Comment (kean) [reply Sources/PulseUI/Features/Console/ConsoleView.swift:17]: I'm not sure it works on macOS, but it does work according to the documentation on other platforms.

**Pull request merged**